include/AutomatableModel.h

```diff
@@ -148,7 +149,18 @@ class LMMS_EXPORT AutomatableModel : public Model, public JournallingObject
 	template<class T>
 	inline T value( int frameOffset = 0 ) const
 	{
-		if( hasLinkedModels() || m_controllerConnection != NULL )
+		if (m_controllerConnection)
+		{
+			if (!m_useControllerValue)
+			{
+				return castValue<T>(m_value);
+			}
+			else
+			{
+				return castValue<T>(controllerValue(frameOffset));
+			}
+		}
+		else if (hasLinkedModels())
```

Comment:
If hasLinkedModels() is true, m_controllerConnection is true and m_useControllerValue is false, it's not the same. The current code returns m_value and your suggestion controllervalue().

Reply:
True, didn't think of that situation. It does behave a little differently than the current code right? I think now automation values have precedence over linked models when there's a controller connected. That seems to be intended from what you commented on the `if (!m_controllerConnection)` conditional though, and I think it's the best alternative since `controllerValue` would return the controller value and not the linked model value in that case. Resolving this for now too.